packages/api/src/external/commonwell/proxy/cw-process-request.ts

```diff
@@ -187,3 +180,60 @@ function replaceAttachmentURL(url: string): string {
   params.append(docContributionFileParam, theURL.pathname);
   return `${docContributionURL}?${params.toString()}`;
 }
+
+export function applyFilterParams(
+  docRefs: DocumentReference[],
+  params: URLSearchParams
+): DocumentReference[] {
+  const date = params.get("date");
+  const countParam = params.get("count");
+  const status = params.get("status");
+
+  let filteredDocRefs = docRefs;
+
+  if (date) {
+    const match = date.match(/(eq|ne|lt|gt|ge|le|sa|eb|ap)?(.*)/);
+    if (match) {
+      const [, prefix, dateString] = match;
+      const filterDate = dayjs(dateString);
+
+      filteredDocRefs = filteredDocRefs.filter(docRef => {
+        const docDate = dayjs(docRef.date);
+
+        switch (prefix) {
+          case "eq":
+            return docDate.isSame(filterDate);
+          case "ne":
+            return !docDate.isSame(filterDate);
+          case "lt":
+            return docDate.isBefore(filterDate);
+          case "gt":
+            return docDate.isAfter(filterDate);
+          case "ge":
+            return docDate.isSame(filterDate) || docDate.isAfter(filterDate);
+          case "le":
+            return docDate.isSame(filterDate) || docDate.isBefore(filterDate);
+          case "sa":
+            return docDate.isAfter(filterDate);
+          case "eb":
+            return docDate.isBefore(filterDate);
```

Comment:
Aren't those:
- `sa`: same or after
- `eb`: equal or before

...in which case we'd need to add the check for sameness, right?